src/schema.ts

```diff
@@ -229,7 +229,14 @@ export default abstract class BaseSchema<
     let base = this;
     let combined = schema.clone();
 
-    const mergedSpec = { ...base.spec, ...combined.spec };
+    const baseSpec = { ...base.spec };
+    // only overwrite with spec values which are not undefined
+    const mergedSpec = Object.keys(combined.spec).reduce((accum, key) => {
```

Comment:
i think it probably simpler to not define undefined properties in the spec instead and keep the object spread

Reply:
Yes, that was the other option, i wasn't really sure about editing that part, it may depend on other stuff as well, not too familiar with yup.